### ISSUE TITLE
feat: Add class imbalance detection warning to train_test_split

### ISSUE THREAD

**Issue by Nsanjayboruds** [probabl-ai/skore#2334]
### Is your feature request related to a problem? Please describe.

Yes. When users split imbalanced datasets without stratification, the test set can have a dramatically different class distribution than the training set, leading to misleading evaluation metrics.

### Example Problem:
```python
from sklearn.datasets import make_classification
from skore import train_test_split

# Create imbalanced dataset (90% class 0, 10% class 1)
X, y = make_classification(n_samples=1000, weights=[0.9, 0.1], random_state=42)

# Without stratification - no warning
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Result:
# - Train set: 90% class 0, 10% class 1
# - Test set: 85% class 0, 15% class 1  ← Different distribution!
# - Evaluation metrics become unreliable
```

### Describe the solution you'd like

Add a warning to `skore.train_test_split()` that:
1. Detects when class distribution differs by >10% between train/test sets
2. Alerts users when they haven't used `stratify=y` with imbalanced data
3. Provides actionable recommendation to use `stratify=y`

### Expected Output:
```python
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# ⚠️ Warning: Class imbalance detected!
# Train set: 90% class 0, 10% class 1
# Test set: 85% class 0, 15% class 1
# Recommendation: Use stratify=y to maintain class distribution
```

### Describe alternatives you've considered, if relevant

**Alternative 1: Silent fix without warning**
- Automatically use stratify when imbalance detected
- Downside: Hides the issue from users, removes transparency

**Alternative 2: Error instead of warning**
- Raise exception when imbalance detected
- Downside: Too strict, breaks existing code

**Selected: Warning approach**
- Alerts users but doesn't break code
- Provides learning opportunity
- Users can choose to use stratify or accept the difference
- Consistent with existing skore warning system

### Why This Matters
Class imbalance is common in real-world datasets (fraud detection, disease diagnosis, etc.). Users often forget stratification, leading to invalid evaluations.

### Related Existing Warnings
The project already has similar warnings:
- `high_class_imbalance_warning.py` - detects severe imbalance
- `stratify_is_set_warning.py` - checks if stratify is used
- This warning fills the gap: detecting distribution mismatch

### Reference
- Related to scikit-learn's stratification best practices
- Complements existing warning infrastructure
- Would help prevent common issues mentioned in ML best practice guides

### Test Scenarios
- Binary classification (2 classes)
- Multiclass (3+ classes)
- Balanced data (no warning)
- Regression (no warning)
- With/without stratify parameter

**Comment by auguste-probabl** [comment]
Thanks for the issue!

Note that skore implements a `StratifyWarning` that fires if `train_test_split` is used *with* the `stratify` argument. With your new proposed check, the user would apply the recommendation to use `stratify`, only to get the `StratifyWarning` right after, which would be poor UX.

If we implement this new check, we'll have to give the user a way out. Maybe we could revisit `StratifyWarning`.

**Comment by Nsanjayboruds** [comment]
Thanks for the detailed feedback, that makes sense 

I agree that triggering a new warning only to immediately raise `StratifyWarning` would be poor UX. One possible direction could be:
- Gate the new imbalance check so it only triggers **when `stratify` is not provided**, and
- Either:
  - silence/deprecate `StratifyWarning` when the imbalance warning already fired, or
  - merge the logic so class imbalance + stratification guidance is handled in a single place.
I’m happy to explore adjusting or refactoring `StratifyWarning` if that’s the preferred direction, or align the behavior with whatever UX you think fits skore best. Let me know how you’d like me to proceed.

**Comment by auguste-probabl** [comment]
Closing as `train_test_split` no longer exists
